sw/device/tests/rv_core_ibex_rnd_test.c

```diff
@@ -0,0 +1,93 @@
+// Copyright lowRISC contributors.
+// Licensed under the Apache License, Version 2.0, see LICENSE for details.
+// SPDX-License-Identifier: Apache-2.0
+
+#include "sw/device/lib/base/memory.h"
+#include "sw/device/lib/base/mmio.h"
+#include "sw/device/lib/dif/dif_edn.h"
+#include "sw/device/lib/dif/dif_rv_core_ibex.h"
+#include "sw/device/lib/runtime/log.h"
+#include "sw/device/lib/testing/entropy_testutils.h"
+#include "sw/device/lib/testing/test_framework/check.h"
+#include "sw/device/lib/testing/test_framework/ottf_main.h"
+
+#include "hw/top_earlgrey/sw/autogen/top_earlgrey.h"
+#include "rv_core_ibex_regs.h"
+
+// Initialize OTTF
+OTTF_DEFINE_TEST_CONFIG();
+
+// Declare two assembly function defined in rv_core_ibex_rnd_test.S
+extern volatile uint32_t load_from_two_addresses(volatile uint32_t *address1,
+                                                 volatile uint32_t *address2);
+extern volatile uint32_t interleave_loads_from_two_addresses(
+    volatile uint32_t *address1, volatile uint32_t *address2);
+
+#define RANDOM_DATA_READS 32
+#define MAX_STATUS_CHECKS 1024
+
+static void wait_till_status_is_valid(const dif_rv_core_ibex_t *rv_core_ibex) {
+  int j = 0;
+  dif_rv_core_ibex_rnd_status_t rnd_status = 0;
+  CHECK(rv_core_ibex != NULL);
```

Comment:
I would drop this checking, since all difs check the handle before using it. So You would get a nice error message in the line 35 anyway.